File: 2024/day_02/solution.ts
(() => {
  const frame1: string[] = ["midu", "madeval", "educalvolpz"];
  const frame2: string[] = ["midu"];
  const frame3: string[] = ["a", "bb", "ccc"];

  function createFrame(names: string[]): string {
    const longestName: number = Math.max(...names.map((name) => name.length));

    const border: string = "*".repeat(longestName + 4);
    const result: string[] = [border];

    names.forEach((name) => {
      const spaces: string = " ".repeat(longestName - name.length);
      result.push(`* ${name}${spaces} *`);
    });

    result.push(border);

    return result.join("\n");
  }

  console.log(createFrame(frame1));
  console.log(createFrame(frame2));
  console.log(createFrame(frame3));
})();

File: 2024/day_02/solution.py
def create_frame(names: list[str]) -> str:
  # Get the longest name in the list
  longest_name = max(len(name) for name in names) if names else 0

  # Create the top border
  border = "*" * (longest_name + 4)
  result = [border]

  # Generate the lines with names and necessary spaces
  for name in names:
    spaces = " " * (longest_name - len(name))
    result.append(f"* {name}{spaces} *")

  # Add the bottom border
  result.append(border)

  # Join the result into a single string separated by line breaks
  return "\n".join(result)
